msal/application.py

```diff
@@ -1207,7 +1207,9 @@ def _acquire_token_silent_from_cache_and_possibly_refresh_it(
             if (result and "error" not in result) or (not access_token_from_cache):
                 return result
         except:  # The exact HTTP exception is transportation-layer dependent
-            logger.exception("Refresh token failed")  # Potential AAD outage?
+            # Typically network error. Potential AAD outage?
+            if not access_token_from_cache:  # It means there is no fall back option
```

Comment:
It is possible for `access_token_from_cache` to be not-`None` here? It seems L1198 will prevent a not-`None` `access_token_from_cache` to reach here.

Reply:
Line 1198 would be skipped by line 1196. So, this check here is still beneficial.

Reply:
Ah, I see. Even if `AT_AGING` refresh fails, `access_token_from_cache` can serve as a fallback. Thanks for the explanation.